os/fs/smartfs/smartfs_smart.c

```diff
@@ -1283,6 +1286,40 @@ static int smartfs_dup(FAR const struct file *oldp, FAR struct file *newp)
 }
 
 /****************************************************************************
+ * Name: smartfs_fstat
+ *
+ * Description:
+ *   Obtain information about an open file associated with the file
+ *   descriptor 'fd', and will write it to the area pointed to by 'buf'.
+ *
+ ****************************************************************************/
+
+static int smartfs_fstat(FAR const struct file *filep, FAR struct stat *buf)
+{
+	FAR struct inode *inode;
+	FAR struct smartfs_mountpt_s *fs;
+	FAR struct smartfs_ofile_s *sf;
+
+	DEBUGASSERT(filep != NULL);
+	DEBUGASSERT(buf != NULL);
+
+	/* Recover our private data from the struct file instance */
+	DEBUGASSERT(filep->f_priv != NULL && filep->f_inode != NULL);
```

Comment:
Like a romfs_fstat, check fileep first then buf.  also move comment to bottom of DEBUGASSERT.

Reply:
Done.